frappe/contacts/report/addresses_and_contacts/test_addresses_and_contacts.py

```diff
@@ -0,0 +1,111 @@
+import frappe
+import frappe.defaults
+import unittest
+
+from frappe.contacts.report.addresses_and_contacts.addresses_and_contacts import get_data
+
+def get_custom_linked_doctype():
+    if bool(frappe.get_all("DocType", filters={'name':'Test Custom Doctype'})):
+        return
+	
+    doc = frappe.get_doc({
+			"doctype": "DocType",
+			"module": "Core",
+			"custom": 1,
+			"fields": [{
+				"label": "Test Field",
+				"fieldname": "test_field",
+				"fieldtype": "Data"
+			},
+			{
+				"label": "Contact HTML",
+				"fieldname": "contact_html",
+				"fieldtype": "HTML"
+			},
+            {
+				"label": "Address HTML",
+				"fieldname": "address_html",
+				"fieldtype": "HTML"
+			}],
+			"permissions": [{
+				"role": "System Manager",
+				"read": 1
+			}],
+			"name": "Test Custom Doctype",
+		}).insert()
+	
+    return doc
+
+def get_custom_doc_for_address_and_contacts():
+	custom_doctype = get_custom_linked_doctype()
+	linked_doc = frappe.get_doc({
+		"doctype": "Test Custom Doctype",
+		"test_field": "Hello",
+		}).insert()
+	return linked_doc
+
+def create_linked_address(link_list):
+    if frappe.flags.test_address_created:
+	    return
+	
+    address = frappe.get_doc({
+		"doctype": "Address",
+		"address_title": "_Test Address",
+        "address_type": "Billing",
```

Comment:
indenting

Reply:
Noted. Sorry for this. Probably messed up my VSCode setup, will ensure this does not happen again.